### PR TITLE
Add support for configuring log level

### PR DESCRIPTION
**Pull request by askore**
Manual testing: check out and deploy https://github.com/acquia/fn-memcached-operator/pull/78 and https://github.com/acquia/fn-drupal-operator/pull/516 and run any test e.g. TestMemcached with no cleanup. Examine the logs of the command pods that run during that test, especially main and mcrouter. The kubexit messages will be gone. You can also change KUBEXIT_LOG_LEVEL to info to see the new format of the info-level messages.

https://core.cloudbees.ais.acquia.io/polaris-jenkins/job/polaris-e2e/3290/

### CODE REVIEW

**Review by blkperl — APPROVED** [review]

File: cmd/kubexit/main.go [hunk 1]
```diff
@@ -47,74 +45,74 @@ func main() {
...
 		}
 	}
-	log.Printf("Grace Period: %s\n", gracePeriod)
+	log.Info("Grace Period:", "grace period", gracePeriod)
 
 	podName := os.Getenv("KUBEXIT_POD_NAME")
 	if podName == "" {
 		if len(birthDeps) > 0 {
-			log.Println("Error: missing env var: KUBEXIT_POD_NAME")
+			log.Error(errors.New("missing env var: KUBEXIT_POD_NAME"), "Error: missing env var: KUBEXIT_POD_NAME")
```
Comment (alexxed) [inline cmd/kubexit/main.go:99]:
```suggestion
			log.Error(errors.New("missing env var: KUBEXIT_POD_NAME"), "missing env var", "var_name", "KUBEXIT_POD_NAME)
```

structured logging allows you to easily filter when searching in logs

File: pkg/log/log.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+var logger = Logger()
+
+func Logger() logr.Logger {
+	l := os.Getenv("KUBEXIT_LOG_LEVEL")
+	level, ok := levelMap[l]
+	if !ok {
+		level = zapcore.InfoLevel
+	}
+
+	log.SetLogger(zap.New(zap.Level(level)))
```
Comment (alexxed) [inline pkg/log/log.go:32]: I haven't tested this, but afaik, zap.New returns a development logger which is not in json format, so you may need to use NewProduction instead https://github.com/uber-go/zap#quick-start

**Review by alexxed — COMMENTED** [review]
This looks good; for better flexibility, add another environment variable that controls the logger profile, development or production https://pkg.go.dev/go.uber.org/zap#hdr-Configuring_Zap; we do that in our operators to have json logs in production.

File: pkg/log/log.go (hunk 2 above)
Comment (renishb10) [reply pkg/log/log.go:32]: Tested Info Level:
```
mcrouter {"level":"info","ts":1625553023.112102,"logger":"kubexit","msg":"Child Exited(0)"}
 mcrouter {"level":"info","ts":1625553023.112142,"logger":"kubexit","msg":"Updating tombstone:","path":"/graveyard/mcrouter"}
 mcrouter stream closed
 main {"level":"info","ts":1625553014.56097,"logger":"kubexit","msg":"Name:","name":"main"}
 main {"level":"info","ts":1625553014.5610745,"logger":"kubexit","msg":"Graveyard:","graveyard":"/graveyard"}
 main {"level":"info","ts":1625553014.5611012,"logger":"kubexit","msg":"Tombstone:","tombstone":"/graveyard/main"}
 main {"level":"info","ts":1625553014.5611215,"logger":"kubexit","msg":"Birth Deps:","birth deps":"mcrouter"}
 main {"level":"info","ts":1625553014.5611386,"logger":"kubexit","msg":"Death Deps: N/A"}
 main {"level":"info","ts":1625553014.5611424,"logger":"kubexit","msg":"Birth Timeout:","birth timeout":30}
 main {"level":"info","ts":1625553014.561163,"logger":"kubexit","msg":"Grace Period:","grace period":30}
 main {"level":"info","ts":1625553014.5611804,"logger":"kubexit","msg":"Pod Name:","pod name":"command-inject-fixtures-56khl-pqtfr"}
 main {"level":"info","ts":1625553014.5611854,"logger":"kubexit","msg":"Namespace:","namespace":"testmemcached-1625552852"}
 main {"level":"info","ts":1625553014.6600866,"logger":"kubexit","msg":"Watching pod updates..."}
```

Its in JSON format, I think we are good here.

**Review by renishb10 — APPROVED** [review]
LGTM
```
shared-setup stream closed
 kubexit stream closed
 mcrouter 2021/07/06 06:14:31 Starting: /usr/bin/mcrouter -p 11211 --config-file=/etc/mcrouter/config.json --asynclog-disable --stats-logging-interval=0 --config-dump-root  --debug-fifo-root  --async-dir

 mcrouter 2021/07/06 06:14:31 Waiting for child process to exit...
 mcrouter I0706 06:14:31.972741    11 StandaloneUtils.cpp:448] 40.0.0 mcrouter startup (11)
 mcrouter I0706 06:14:31.972879    11 StandaloneUtils.cpp:186] Starting Memcache router
 mcrouter I0706 06:14:31.972903    11 Server-inl.h:172] Spawning AsyncMcServer
 mcrouter I0706 06:14:31.973273    11 FunctionScheduler.cpp:405] Starting FunctionScheduler with 0 functions.
 mcrouter I0706 06:14:31.974799    11 CarbonRouterInstance-inl.h:516] creating config builder
 mcrouter I0706 06:14:31.976341    11 CarbonRouterInstance-inl.h:487] started reconfiguring
 mcrouter I0706 06:14:31.976661    11 CarbonRouterInstance-inl.h:504] reconfigured 1 proxies with 1 pools, 2 clients 613e9cf13532fa8c7c683eeffd053fc1)
 main Event Type: ADDED
 main Event Type: MODIFIED
```

File: .gitignore [hunk 3]
```diff
@@ -19,3 +19,6 @@
 
 # Test graveyard
 .graveyard
+
+# Editor
+.idea
```
[the remark this answers is not in the document]
Comment (renishb10) [reply .gitignore:24]: fyi: I dont have `.idea` in my .gitignore_global. It looks like by default it has only `.DS_Store` (OSx)

**Pull request merged**